vmdb/lib/workers/worker_base.rb

```diff
@@ -88,16 +90,14 @@ def self.wait_for_worker_monitor=(val)
 
   def my_monitor_started?
     return @monitor_started unless @monitor_started.nil?
-    server = MiqServer.my_server(true)
     return false if     server.nil?
-    return false unless server.started?
+    return false unless server.reload.started?
```

Comment:
This is good... we keep calling `my_monitor_started?` in a loop in `do_wait_for_worker_monitor` until it's started  (in the main thread) so we need to get the current value, not a cached one.  :+1: